tools/ci/download_bin.sh

```diff
@@ -57,39 +68,74 @@ download_artifact_linux()
-    echo -e "\033[32mFinished. Please check ${output_dir}\033[0m"
+    if [[ "${result}" != "0" ]];then LOG_ERROR "Download failed, please try again" && exit 1;fi
 
 }
 
 download_artifact_macOS(){
     echo "unsupported for now."
     exit 1
+    local branch="$1"
     # TODO: https://developer.github.com/v3/actions/artifacts/#download-an-artifact
-    local workflow_artifacts_url="$(curl https://api.github.com/repos/FISCO-BCOS/FISCO-BCOS/actions/runs\?branch\=${branch}\&status\=success\&event\=push | grep artifacts_url | head -n 1 | cut -d \" -f 4)"
+    local workflow_artifacts_url="$(curl https://api.github.com/repos/${project}/actions/runs\?branch\=${branch}\&status\=success\&event\=push | grep artifacts_url | head -n 1 | cut -d \" -f 4)"
+    # local archive_download_url="$(curl https://api.github.com/repos/FISCO-BCOS/FISCO-BCOS/actions/artifacts | grep archive_download_url | sort | tail -n 1 | cut -d \" -f 4)"
     local archive_download_url="$(curl ${workflow_artifacts_url} | grep archive_download_url | cut -d \" -f 4)"
     if [ -z "${archive_download_url}" ];then
         echo "GitHub action ${workflow_artifacts_url} can't find artifact."
         exit 1
     fi
-    echo -e "\033[32mDownloading macOS binary from ${archive_download_url} \033[0m"
-    # FIXME: https://github.com/actions/upload-artifact/issues/51
-    cd "${output_dir}" && curl -LO "${archive_download_url}" && tar -zxf fisco-bcos-macOS.tar.gz && rm fisco-bcos-macOS.tar.gz
+    LOG_INFO "Downloading macOS binary from ${archive_download_url}"
+    # failed because of permission issue
+    # FIXME: https://github.com/actions/upload-artifact/issues/51 https://github.community/t/api-to-download-artifact/16496/13
+    curl -#Lo fisco-bcos-macOS.tar.gz "${archive_download_url}" && tar -zxf fisco-bcos-macOS.tar.gz && rm fisco-bcos-macOS.tar.gz
 }
 
-main(){
-    [ -f "${output_dir}/fisco-bcos" ] && rm -f "${output_dir}/fisco-bcos"
-    mkdir -p "${output_dir}"
+download_branch_artifact(){
+    local branch="$1"
     if [ "$(uname)" == "Darwin" ];then
-        echo -e "\033[32mDownloading binary of macOS \033[0m"
-        download_artifact_macOS
+        LOG_INFO "Downloading binary of ${branch} on macOS "
+        download_artifact_macOS "${branch}"
+    else
+        LOG_INFO "Downloading binary of ${branch} on Linux "
+        download_artifact_linux "${branch}"
+    fi
+}
+
+download_released_artifact(){
+    local version="$1"
+    local package_name="fisco-bcos.tar.gz"
+    if [ "$(uname)" == "Darwin" ];then
+        package_name="fisco-bcos-macOS.tar.gz"
+    fi
+    local download_link="https://github.com/FISCO-BCOS/FISCO-BCOS/releases/download/${version}/${package_name}"
+    local cdn_download_link="${cdn_link_header}/${version}/${package_name}"
+    LOG_INFO "Downloading binary of ${version}, from ${download_link}"
+    if [ $(curl -IL -o /dev/null -s -w %{http_code} ${cdn_download_link}) == 200 ];then
```

Comment:
Codacy found an issue: [Double quote to prevent globbing and word splitting.](https://app.codacy.com/gh/FISCO-BCOS/FISCO-BCOS/pullRequest?prid=5987732)